File: go/1046.go
package leetcode

import "sort"

func lastStoneWeight(stones []int) int {

	for len(stones) > 1 {
		sort.Slice(stones, func(i, j int) bool { return stones[i] > stones[j] })
		max1, max2 := stones[0], stones[1]
		if max1 == max2 {
			stones = append(stones[:0], stones[2:]...)
		} else if max1 != max2 {
			stones[0] = stones[0] - stones[1]
			stones = append(stones[:1], stones[2:]...)
		}
	}

	if len(stones) == 1 {
		return stones[0]
	} else {
		return 0
	}

}

File: javascript/1046.js
/**
 * @param {number[]} stones
 * @return {number}
 */
var lastStoneWeight = function (stones) {
  while (stones.length > 1) {
    r;
    stones.sort((a, b) => b - a);

    let max1 = stones[0];
    let max2 = stones[1];

    if (max1 === max2) {
      stones = stones.slice(2);
    } else {
      stones[0] = max1 - max2;
      stones = stones.slice(0, 1).concat(stones.slice(2));
    }
  }

  return stones.length === 1 ? stones[0] : 0;
};
